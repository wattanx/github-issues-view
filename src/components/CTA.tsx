import NextLink from 'next/link';
import { FaArrowRight } from 'react-icons/fa';
import { DiGithubBadge } from 'react-icons/di';
import { Container, Button, Stack } from 'components';

export const CTA = () => (
  <Container
    flexDirection={['unset', 'row', 'row', 'row']}
    width="100%"
    maxWidth="48rem"
    py={3}
  >
    <Stack
      mt="10"
      spacing="4"
      justify="center"
      direction={{ base: 'column', sm: 'row' }}
    >
      <NextLink href="/issues" passHref>
        <Button
          h="4rem"
          px="40px"
          fontSize="1.2rem"
          as="a"
          size="lg"
          bg="teal.200"
          rightIcon={<FaArrowRight fontSize="0.8em" />}
        >
          Get Started
        </Button>
      </NextLink>

      <Button
        as="a"
        size="lg"
        h="4rem"
        px="40px"
        fontSize="1.2rem"
        href="https://github.com/wattanx/github-issues-view"
        target="__blank"
        leftIcon={<DiGithubBadge size="1.5em" />}
      >
        GitHub
      </Button>
    </Stack>
  </Container>
);
