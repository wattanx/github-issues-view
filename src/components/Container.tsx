import { Box, BoxProps } from 'components';

export const Container = (props: BoxProps) => {
  return (
    <Box
      w="full"
      pb="12"
      pt="3"
      maxW={{ base: 'xl', md: '7xl' }}
      mx="auto"
      px={{ base: '6', md: '8' }}
      {...props}
    />
  );
};
