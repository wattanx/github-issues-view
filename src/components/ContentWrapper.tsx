import { Box } from '../components';

export type ContentWrapperProps = {
  children: React.ReactNode;
};
export const ContentWrapper: React.FC<ContentWrapperProps> = ({ children }) => (
  <Box maxWidth="1280px" margin="auto" padding={5}>
    {children}
  </Box>
);
