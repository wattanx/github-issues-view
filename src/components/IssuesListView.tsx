import NextLink from 'next/link';
import {
  Box,
  ClosedStatus,
  Flex,
  OpenedStatus,
  Stack,
  Text,
} from '../components';
import { IssueStateType } from 'types/github';
import { IssueType } from 'framework';

export const IssuesListView: React.VFC<{ issues?: IssueType[] }> = ({
  issues,
}) => {
  return (
    <Stack spacing="0">
      {issues &&
        issues.map((x) => {
          return (
            <IssuesListCell
              key={x.issueNumber}
              title={x.title}
              issueNumber={x.issueNumber}
              state={x.state}
            />
          );
        })}
    </Stack>
  );
};

type IssuesListCellProps = {
  state: IssueStateType;
  title: string;
  issueNumber: number;
};

const IssuesListCell: React.VFC<IssuesListCellProps> = ({
  state,
  title,
  issueNumber,
}) => (
  <Flex
    borderTopColor="#e2e2e2"
    borderTopWidth="1px"
    _first={{ borderTopColor: 'transparent' }}
  >
    <Box paddingTop={3} paddingLeft={3}>
      {state === 'open' && <OpenedStatus />}
      {state === 'closed' && <ClosedStatus />}
    </Box>
    <Box padding={2}>
      <NextLink
        href="/issues/[issueNumber]"
        as={`/issues/${issueNumber}`}
        shallow={true}
        passHref
      >
        <Box as="a">
          <Text>{title}</Text>
        </Box>
      </NextLink>
      <Text fontSize="sm">#{issueNumber}</Text>
    </Box>
  </Flex>
);
