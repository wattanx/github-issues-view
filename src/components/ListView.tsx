import { Box } from 'components';

export type ListViewProps = {
  children: React.ReactNode;
};

export type ListViewCellProps = {
  children: React.ReactNode;
};

export const ListView: React.FC<ListViewProps> = ({ children }) => (
  <Box>{children}</Box>
);

export const ListViewCell: React.FC<ListViewCellProps> = ({ children }) => (
  <Box>{children}</Box>
);
