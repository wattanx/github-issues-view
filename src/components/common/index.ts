export * from './icon';
export * from './layout';
export * from './typography';
