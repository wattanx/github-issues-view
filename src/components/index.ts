export * from './common';
export * from './Container';
export * from './ContentWrapper';
export * from './Status';
export * from './IssuesListView';
