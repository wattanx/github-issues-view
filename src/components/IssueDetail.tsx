import { Text, Heading, HStack } from '@chakra-ui/react';
import { Box } from './common';

export type IssueDetailProps = {
  title: string;
  body: string;
  state: string;
  issueNumber: string;
};

export const IssueDetail: React.VFC<IssueDetailProps> = ({
  title,
  body,
  state,
  issueNumber,
}) => {
  return (
    <Box>
      <Box borderBottomWidth="1px" borderBottomColor="#e2e2e2">
        <HStack>
          <Heading>
            {title}
            <Box fontWeight="normal" color="#b9b9b9" as="span" marginLeft={2}>
              #{issueNumber}
            </Box>
          </Heading>
        </HStack>
      </Box>
      <Box rounded="md" border="1px #e2e2e2 solid" marginTop={3} padding={2}>
        <Box borderBottom="1px #e2e2e2 solid" padding={1}></Box>
        <Box marginTop={3} width={['100%', '100%', '100%', '70%']}>
          {body}
        </Box>
      </Box>
    </Box>
  );
};
