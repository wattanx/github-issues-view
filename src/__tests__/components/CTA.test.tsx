import { render } from '@testing-library/react';
import { CTA } from 'components';

test('CTA Snapshot Test', () => {
  const component = render(<CTA />);
  expect(component).toMatchSnapshot();
});
