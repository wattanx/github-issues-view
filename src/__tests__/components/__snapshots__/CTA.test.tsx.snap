// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTA Snapshot Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-6rx5i2"
      >
        <div
          class="chakra-stack css-5l6diy"
        >
          <a
            class="chakra-button css-k64uyl"
            href="/issues"
          >
            Get Started
            <span
              class="chakra-button__icon css-1hzyiq5"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                font-size="0.8em"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
                />
              </svg>
            </span>
          </a>
          <a
            class="chakra-button css-qq75xv"
            href="https://github.com/wattanx/github-issues-view"
            target="__blank"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1.5em"
                stroke="currentColor"
                stroke-width="0"
                version="1.1"
                viewBox="0 0 32 32"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 5.343c-6.196 0-11.219 5.023-11.219 11.219 0 4.957 3.214 9.162 7.673 10.645 0.561 0.103 0.766-0.244 0.766-0.54 0-0.267-0.010-1.152-0.016-2.088-3.12 0.678-3.779-1.323-3.779-1.323-0.511-1.296-1.246-1.641-1.246-1.641-1.020-0.696 0.077-0.682 0.077-0.682 1.126 0.078 1.72 1.156 1.72 1.156 1.001 1.715 2.627 1.219 3.265 0.931 0.102-0.723 0.392-1.219 0.712-1.498-2.49-0.283-5.11-1.246-5.11-5.545 0-1.226 0.438-2.225 1.154-3.011-0.114-0.285-0.501-1.426 0.111-2.97 0 0 0.941-0.301 3.085 1.15 0.894-0.25 1.854-0.373 2.807-0.377 0.953 0.004 1.913 0.129 2.809 0.379 2.14-1.453 3.083-1.15 3.083-1.15 0.613 1.545 0.227 2.685 0.112 2.969 0.719 0.785 1.153 1.785 1.153 3.011 0 4.31-2.624 5.259-5.123 5.537 0.404 0.348 0.761 1.030 0.761 2.076 0 1.5-0.015 2.709-0.015 3.079 0 0.299 0.204 0.648 0.772 0.538 4.455-1.486 7.666-5.69 7.666-10.645 0-6.195-5.023-11.219-11.219-11.219z"
                />
              </svg>
            </span>
            GitHub
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-6rx5i2"
    >
      <div
        class="chakra-stack css-5l6diy"
      >
        <a
          class="chakra-button css-k64uyl"
          href="/issues"
        >
          Get Started
          <span
            class="chakra-button__icon css-1hzyiq5"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              font-size="0.8em"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              />
            </svg>
          </span>
        </a>
        <a
          class="chakra-button css-qq75xv"
          href="https://github.com/wattanx/github-issues-view"
          target="__blank"
        >
          <span
            class="chakra-button__icon css-1wh2kri"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1.5em"
              stroke="currentColor"
              stroke-width="0"
              version="1.1"
              viewBox="0 0 32 32"
              width="1.5em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 5.343c-6.196 0-11.219 5.023-11.219 11.219 0 4.957 3.214 9.162 7.673 10.645 0.561 0.103 0.766-0.244 0.766-0.54 0-0.267-0.010-1.152-0.016-2.088-3.12 0.678-3.779-1.323-3.779-1.323-0.511-1.296-1.246-1.641-1.246-1.641-1.020-0.696 0.077-0.682 0.077-0.682 1.126 0.078 1.72 1.156 1.72 1.156 1.001 1.715 2.627 1.219 3.265 0.931 0.102-0.723 0.392-1.219 0.712-1.498-2.49-0.283-5.11-1.246-5.11-5.545 0-1.226 0.438-2.225 1.154-3.011-0.114-0.285-0.501-1.426 0.111-2.97 0 0 0.941-0.301 3.085 1.15 0.894-0.25 1.854-0.373 2.807-0.377 0.953 0.004 1.913 0.129 2.809 0.379 2.14-1.453 3.083-1.15 3.083-1.15 0.613 1.545 0.227 2.685 0.112 2.969 0.719 0.785 1.153 1.785 1.153 3.011 0 4.31-2.624 5.259-5.123 5.537 0.404 0.348 0.761 1.030 0.761 2.076 0 1.5-0.015 2.709-0.015 3.079 0 0.299 0.204 0.648 0.772 0.538 4.455-1.486 7.666-5.69 7.666-10.645 0-6.195-5.023-11.219-11.219-11.219z"
              />
            </svg>
          </span>
          GitHub
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
